packages/flutter/lib/src/rendering/grid.dart

```diff
@@ -5,67 +5,267 @@
 import 'box.dart';
 import 'object.dart';
 
-class _GridMetrics {
-  // Grid is width-in, height-out.  We fill the max width and adjust height
-  // accordingly.
-  factory _GridMetrics({ double width, int childCount, double maxChildExtent }) {
-    assert(width != null);
-    assert(childCount != null);
-    assert(maxChildExtent != null);
-    double childExtent = maxChildExtent;
-    int childrenPerRow = (width / childExtent).floor();
-    // If the child extent divides evenly into the width use that, otherwise + 1
-    if (width / childExtent != childrenPerRow.toDouble()) childrenPerRow += 1;
-    double totalPadding = 0.0;
-    if (childrenPerRow * childExtent > width) {
-      // TODO(eseidel): We should snap to pixel bounderies.
-      childExtent = width / childrenPerRow;
-    } else {
-      totalPadding = width - (childrenPerRow * childExtent);
-    }
-    double childPadding = totalPadding / (childrenPerRow + 1.0);
-    int rowCount = (childCount / childrenPerRow).ceil();
+bool _isMonotonic(List<double> offsets) {
```

Comment:
You only call this in asserts, so we should probably call it _debugIsMonotonic() and wrap its contents in an assert, to lower the risk of anyone copy-and-pasting it into release mode.